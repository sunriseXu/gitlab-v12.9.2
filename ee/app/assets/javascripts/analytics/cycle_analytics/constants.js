import { __ } from '~/locale';

export const PROJECTS_PER_PAGE = 50;

export const DEFAULT_DAYS_IN_PAST = 30;

export const EVENTS_LIST_ITEM_LIMIT = 50;

export const EMPTY_STAGE_TEXT = {
  issue: __(
    'The issue stage shows the time it takes from creating an issue to assigning the issue to a milestone, or add the issue to a list on your Issue Board. Begin creating issues to see data for this stage.',
  ),
  plan: __(
    'The planning stage shows the time from the previous step to pushing your first commit. This time will be added automatically once you push your first commit.',
  ),
  code: __(
    'The coding stage shows the time from the first commit to creating the merge request. The data will automatically be added here once you create your first merge request.',
  ),
  test: __(
    'The testing stage shows the time GitLab CI takes to run every pipeline for the related merge request. The data will automatically be added after your first pipeline finishes running.',
  ),
  review: __(
    'The review stage shows the time from creating the merge request to merging it. The data will automatically be added after you merge your first merge request.',
  ),
  staging: __(
    'The staging stage shows the time between merging the MR and deploying code to the production environment. The data will be automatically added once you deploy to production for the first time.',
  ),
  production: __(
    'The production stage shows the total time it takes between creating an issue and deploying the code to production. The data will be automatically added once you have completed the full idea to production cycle.',
  ),
};

export const DEFAULT_STAGE_NAMES = [...Object.keys(EMPTY_STAGE_TEXT), 'total'];

export const TASKS_BY_TYPE_SUBJECT_ISSUE = 'Issue';
export const TASKS_BY_TYPE_SUBJECT_MERGE_REQUEST = 'MergeRequest';
export const TASKS_BY_TYPE_MAX_LABELS = 15;

export const TASKS_BY_TYPE_SUBJECT_FILTER_OPTIONS = {
  [TASKS_BY_TYPE_SUBJECT_ISSUE]: __('Issues'),
  [TASKS_BY_TYPE_SUBJECT_MERGE_REQUEST]: __('Merge Requests'),
};

export const TASKS_BY_TYPE_FILTERS = {
  SUBJECT: 'SUBJECT',
  LABEL: 'LABEL',
};

export const STAGE_ACTIONS = {
  SELECT: 'selectStage',
  EDIT: 'editStage',
  REMOVE: 'removeStage',
  HIDE: 'hideStage',
  CREATE: 'createStage',
  UPDATE: 'updateStage',
};

export const STAGE_NAME = {
  TOTAL: 'total',
  PRODUCTION: 'production',
};
