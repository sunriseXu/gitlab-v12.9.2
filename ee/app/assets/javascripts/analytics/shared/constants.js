export const dateFormats = {
  isoDate: 'yyyy-mm-dd',
  defaultDate: 'mmm d, yyyy',
  defaultDateTime: 'mmm d, yyyy h:MMtt',
};

export const scatterChartLineProps = {
  default: {
    type: 'line',
    showSymbol: false,
    // By default zlevel is 2 for all series types.
    // By increasing the zlevel to 3 we make sure that the trendline gets drawn in front of the dots in the chart.
    zlevel: 3,
  },
};

export const LAST_ACTIVITY_AT = 'last_activity_at';

export const DATE_RANGE_LIMIT = 180;

export const OFFSET_DATE_BY_ONE = 1;
