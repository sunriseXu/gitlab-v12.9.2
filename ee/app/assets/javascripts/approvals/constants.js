import { __ } from '~/locale';

export const TYPE_USER = 'user';
export const TYPE_GROUP = 'group';
export const TYPE_HIDDEN_GROUPS = 'hidden_groups';

export const RULE_TYPE_FALLBACK = 'fallback';
export const RULE_TYPE_REGULAR = 'regular';
export const RULE_TYPE_REPORT_APPROVER = 'report_approver';
export const RULE_TYPE_CODE_OWNER = 'code_owner';
export const RULE_TYPE_ANY_APPROVER = 'any_approver';
export const RULE_NAME_ANY_APPROVER = 'All Members';

export const VULNERABILITY_CHECK_NAME = 'Vulnerability-Check';
export const LICENSE_CHECK_NAME = 'License-Check';

export const APPROVAL_RULE_CONFIGS = {
  [VULNERABILITY_CHECK_NAME]: {
    title: __('Vulnerability-Check'),
    popoverText: __(
      'A merge request approval is required when a security report contains a new vulnerability of high, critical, or unknown severity.',
    ),
    documentationText: __('Learn more about Vulnerability-Check'),
  },
  [LICENSE_CHECK_NAME]: {
    title: __('License-Check'),
    popoverText: __(
      'A merge request approval is required when the license compliance report contains a blacklisted license.',
    ),
    documentationText: __('Learn more about License-Check'),
  },
};
