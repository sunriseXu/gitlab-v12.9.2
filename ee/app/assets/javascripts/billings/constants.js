export const TABLE_TYPE_DEFAULT = 'default';
export const TABLE_TYPE_FREE = 'free';
export const TABLE_TYPE_TRIAL = 'trial';
