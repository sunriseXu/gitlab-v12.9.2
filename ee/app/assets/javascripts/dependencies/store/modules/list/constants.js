import { __, s__ } from '~/locale';

export const SORT_FIELDS = {
  name: s__('Dependencies|Component name'),
  packager: s__('Dependencies|Packager'),
  severity: s__('Vulnerability|Severity'),
};

export const SORT_ORDER = {
  ascending: 'asc',
  descending: 'desc',
};

export const REPORT_STATUS = {
  ok: 'ok',
  jobNotSetUp: 'job_not_set_up',
  jobFailed: 'job_failed',
  noDependencies: 'no_dependencies',
  incomplete: 'no_dependency_files',
};

export const FILTER = {
  all: 'all',
  vulnerable: 'vulnerable',
};

export const FETCH_ERROR_MESSAGE = __(
  'Error fetching the dependency list. Please check your network connection and try again.',
);
