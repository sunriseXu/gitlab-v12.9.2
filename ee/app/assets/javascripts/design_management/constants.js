// WARNING: replace this with something
// more sensical as per https://gitlab.com/gitlab-org/gitlab/issues/118611
export const VALID_DESIGN_FILE_MIMETYPE = {
  mimetype: 'image/*',
  regex: /image\/.+/,
};

// https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/types
export const VALID_DATA_TRANSFER_TYPE = 'Files';
