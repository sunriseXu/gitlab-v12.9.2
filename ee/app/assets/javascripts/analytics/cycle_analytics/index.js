import Vue from 'vue';
import CycleAnalytics from './components/base.vue';
import createStore from './store';
import { buildCycleAnalyticsInitialData } from '../shared/utils';
import { parseBoolean } from '~/lib/utils/common_utils';

export default () => {
  const el = document.querySelector('#js-cycle-analytics-app');
  const { emptyStateSvgPath, noDataSvgPath, noAccessSvgPath, hideGroupDropDown } = el.dataset;

  const initialData = buildCycleAnalyticsInitialData(el.dataset);
  const store = createStore();
  store.dispatch('initializeCycleAnalytics', initialData);

  return new Vue({
    el,
    name: 'CycleAnalyticsApp',
    store,
    render: createElement =>
      createElement(CycleAnalytics, {
        props: {
          emptyStateSvgPath,
          noDataSvgPath,
          noAccessSvgPath,
          hideGroupDropDown: parseBoolean(hideGroupDropDown),
        },
      }),
  });
};
