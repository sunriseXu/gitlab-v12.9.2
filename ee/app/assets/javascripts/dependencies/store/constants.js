import { s__ } from '~/locale';
import { FILTER } from './modules/list/constants';

// eslint-disable-next-line import/prefer-default-export
export const DEPENDENCY_LIST_TYPES = {
  all: {
    namespace: 'allDependencies',
    label: s__('Dependencies|All'),
    initialState: {
      filter: FILTER.all,
    },
  },
  vulnerable: {
    namespace: 'vulnerableDependencies',
    label: s__('Dependencies|Vulnerable components'),
    initialState: {
      filter: FILTER.vulnerable,
    },
  },
};
