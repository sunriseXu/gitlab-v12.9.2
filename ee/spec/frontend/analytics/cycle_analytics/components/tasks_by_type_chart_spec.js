import { mount, shallowMount } from '@vue/test-utils';
import TasksByTypeChart from 'ee/analytics/cycle_analytics/components/tasks_by_type_chart.vue';
import { TASKS_BY_TYPE_SUBJECT_ISSUE } from 'ee/analytics/cycle_analytics/constants';
import { groupLabels } from '../mock_data';

const seriesNames = ['Cool label', 'Normal label'];
const data = [[0, 1, 2], [5, 2, 3], [2, 4, 1]];
const groupBy = ['Group 1', 'Group 2', 'Group 3'];
const filters = {
  selectedGroup: {
    id: 22,
    name: 'Gitlab Org',
    fullName: 'Gitlab Org',
    fullPath: 'gitlab-org',
  },
  selectedProjectIds: [],
  startDate: new Date('2019-12-11'),
  endDate: new Date('2020-01-10'),
  subject: TASKS_BY_TYPE_SUBJECT_ISSUE,
  selectedLabelIds: [1, 2, 3],
};

function createComponent({ props = {}, shallow = true, stubs = {} }) {
  const fn = shallow ? shallowMount : mount;
  return fn(TasksByTypeChart, {
    propsData: {
      filters,
      chartData: {
        groupBy,
        data,
        seriesNames,
      },
      labels: groupLabels,
      ...props,
    },
    stubs: {
      'gl-stacked-column-chart': true,
      'tasks-by-type-filters': true,
      ...stubs,
    },
  });
}

describe('TasksByTypeChart', () => {
  let wrapper = null;

  afterEach(() => {
    wrapper.destroy();
  });

  describe('with data available', () => {
    beforeEach(() => {
      wrapper = createComponent({});
    });

    it('should render the loading chart', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('no data available', () => {
    beforeEach(() => {
      wrapper = createComponent({
        props: {
          chartData: {
            groupBy: [],
            data: [],
            seriesNames: [],
          },
        },
      });
    });

    it('should render the no data available message', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
