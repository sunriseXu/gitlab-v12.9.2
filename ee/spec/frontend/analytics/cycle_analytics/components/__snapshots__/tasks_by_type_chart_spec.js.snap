// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksByTypeChart no data available should render the no data available message 1`] = `
"<div class=\\"row\\">
  <div class=\\"col-12\\">
    <h3>Type of work</h3>
    <div class=\\"bs-callout bs-callout-info\\">
      <p>There is no data available. Please change your selection.</p>
    </div>
  </div>
</div>"
`;

exports[`TasksByTypeChart with data available should render the loading chart 1`] = `
"<div class=\\"row\\">
  <div class=\\"col-12\\">
    <h3>Type of work</h3>
    <div>
      <p>Showing data for group 'Gitlab Org' from Dec 11, 2019 to Jan 10, 2020</p>
      <tasks-by-type-filters-stub maxlabels=\\"15\\" labels=\\"[object Object],[object Object],[object Object]\\" selectedlabelids=\\"1,2,3\\" subjectfilter=\\"Issue\\"></tasks-by-type-filters-stub>
      <gl-stacked-column-chart-stub data=\\"0,1,2,5,2,3,2,4,1\\" option=\\"[object Object]\\" presentation=\\"stacked\\" groupby=\\"Group 1,Group 2,Group 3\\" xaxistype=\\"category\\" xaxistitle=\\"Date\\" yaxistitle=\\"Number of tasks\\" seriesnames=\\"Cool label,Normal label\\" legendaveragetext=\\"Avg\\" legendmaxtext=\\"Max\\" y-axis-type=\\"value\\"></gl-stacked-column-chart-stub>
    </div>
  </div>
</div>"
`;
